### DISCUSSION TITLE
Mapping proximity sensor

### DISCUSSION THREAD

**Discussion by xoanxil** [tttapa/Control-Surface#824] [Q&A]
Hi,

I am trying to use a ultrared proximity sensor. I get out of range values with this standard code:

#include <Control_Surface.h> // Include the Control Surface library

// Instantiate a MIDI over USB interface.
USBMIDI_Interface midi;

CCPotentiometer knobsTop[] {
{A0, MIDI_CC::General_Purpose_Controller_1}};

void setup() {
  Control_Surface.begin(); // Initialize Control Surface
}

void loop() {
  Control_Surface.loop(); // Update the Control Surface
}

Couls someone help me mapping this value into 0- 127? I know you can use map() in Arduino but I have made some test and I don´t know how to aplly this with Control-Surface library.

Thanks in advance.

Best

**Comment by tttapa** [comment]
> I am trying to use a ultrared proximity sensor. I get out of range values with this standard code:

Unlikely, the ADC cannot read out-of-range values, and MIDI cannot transmit them.  
What values do you get, and what values do you expect?

Ton answer the question about mapping: see [CCPotentiometer-Map.ino](https://tttapa.github.io/Control-Surface-doc/Doxygen/d3/d26/CCPotentiometer-Map_8ino-example.html)

**Comment by xoanxil** [comment]
Thanks for you quick answer. 

If I map directly in a DAW I get values between 2 and 65 aprox. and I would like to map into 0-127 to standard values to control a full dial or fader.

**Reply by tttapa** [reply]
Yes, you can do that using the CCPotentiometer-Map example.

**Comment by bobo553** [comment]
这是一封自动回复邮件。已经收到您的来信，我会尽快回复。

**Comment by xoanxil** [comment]
Thanks for your answers!
